src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2236,6 +2237,7 @@ private function registerNotifierConfiguration(array $config, ContainerBuilder $
             AllMySmsTransportFactory::class => 'notifier.transport_factory.allmysms',
             FirebaseTransportFactory::class => 'notifier.transport_factory.firebase',
             FreeMobileTransportFactory::class => 'notifier.transport_factory.freemobile',
+            FakeSmsTransportFactory::class => 'notifier.transport_factory.fakesms',
```

Comment:
shouldnt this fake transport only be configured for `dev` env?

also i really like the idea of having a fake transport, i have a question; why to rely on a mailer mail? why not a monolog log?
they are displayed in console, they are displayed in profiler, they are stored in file and it does not need another software to be installed

could not found the related issue that lead to this PR :)
thank you

Reply:
> shouldnt this fake transport only be configured for dev env?
I agree with that too.

> why to rely on a mailer mail? why not a monolog log?
IMOH, I think it's will great if we can choose the destination.

For example :
 - By email if the content contains a required information for the workflow (otp code)
 - By log if the content is just an information and have no importance
 - In a database or a redis to use that in functionnals testing and we should check or get information in the content message

 What do you think ?

Reply:
@ke20 `What do you think ?`

I do not know the feature request/needs that initiate this PR, as written upper :)
But iiuc, the aim is to fake, in dev, the notifier sms logic to see if it works

I think that relying only on the mailer is overkill, a logger is enough => thus my previous comment :)

maybe yes a configured fake transport with multiple sublogic (log > mailer > database) can be the way, lets wait other reviews

Reply:
I agree, it would be interesting to be able to use other distribution channels. The packet was originally developed for the development of an otp system, important in the user workflow, so the choice of emails was made to facilitate testing.

Also, support of chat notifier would be a good thing.

I could do the integration of other channels in a few weeks, if anyone would like to contribute to this PR, feel free.